example_publisher/providers/coin_gecko.py

```diff
@@ -45,18 +47,17 @@ def _update_prices(self) -> None:
             ids=list(self._prices.keys()), vs_currencies=USD, precision=18
         )
         for id_, prices in result.items():
-            self._prices[id_] = prices[USD]
+            price = prices[USD]
+            self._prices[id_] = Price(
+                price,
+                price * self._config.confidence_ratio_bps / 10000,
+                floor(time.time()),
+            )
         log.info("updated prices from CoinGecko", prices=self._prices)
 
     def _get_price(self, id: Id) -> float:
```

Comment:
the return value here will be Price based on your change

Reply:
`Optional[Price]` right?

Reply:
Fixed